tests/Microsoft.Identity.Test.Unit/ApiConfigTests/AuthorityTests.cs

```diff
@@ -33,75 +54,107 @@ public void VerifyAuthorityTest()
+             resultTid: utid,
+             _testRequestContext);
 
             VerifyAuthority(
              config: s_commonAuthority,
              request: s_utid2Authority,
              accountTid: utid,
-             resultTid: utid2);
+             resultTid: utid2,
+             _testRequestContext);
         }
 
         [TestMethod]
-        public void AuthorityMismatchTest()
+        public void AuthorityMismatchTestAsync()
         {
-            var ex = AssertException.Throws<MsalClientException>(() =>
-                Authority.CreateAuthorityForRequest(s_utidAuthority, s_b2cAuthority, null));
+            _testRequestContext.ServiceBundle.Config.AuthorityInfo = s_utidAuthority;
+            var ex =  AssertException.Throws<MsalClientException>(
+                                             () => Authority.CreateAuthorityForRequestAsync(_testRequestContext, s_b2cAuthority, null));
 
             Assert.AreEqual(MsalError.AuthorityTypeMismatch, ex.ErrorCode);
         }
 
         [TestMethod]
         public void DefaultAuthorityDifferentTypeTest()
         {
-            var ex = Assert.ThrowsException<MsalClientException>(
-                () => Authority.CreateAuthorityForRequest(s_commonAuthority, s_b2cAuthority, null));
+            _testRequestContext.ServiceBundle.Config.AuthorityInfo = s_commonAuthority;
+            var ex = Assert.ThrowsException<AggregateException>(
+                () => Authority.CreateAuthorityForRequestAsync(_testRequestContext, s_b2cAuthority, null).Result);
 
-            Assert.AreEqual(MsalError.B2CAuthorityHostMismatch, ex.ErrorCode);
+            Assert.AreEqual(MsalError.B2CAuthorityHostMismatch, (ex.InnerException as MsalClientException).ErrorCode);
         }
 
         [TestMethod]
         public void DifferentHosts()
         {
-            var ex = Assert.ThrowsException<MsalClientException>(
-                () => Authority.CreateAuthorityForRequest(s_commonAuthority, s_ppeAuthority, null));
-            Assert.AreEqual(MsalError.AuthorityHostMismatch, ex.ErrorCode);
-
-            var ex2 = Assert.ThrowsException<MsalClientException>(
-              () => Authority.CreateAuthorityForRequest(s_ppeAuthority, s_commonAuthority, null));
-            Assert.AreEqual(MsalError.AuthorityHostMismatch, ex2.ErrorCode);
-
-            var ex3 = Assert.ThrowsException<MsalClientException>(
-             () => Authority.CreateAuthorityForRequest(
-                 AuthorityInfo.FromAdfsAuthority(TestConstants.ADFSAuthority, true),
+            _harness.HttpManager.AddInstanceDiscoveryMockHandler();
+            _testRequestContext.ServiceBundle.Config.HttpManager = _harness.HttpManager;
+            _testRequestContext.ServiceBundle.Config.AuthorityInfo = s_commonAuthority;
+            var ex = Assert.ThrowsException<AggregateException>(
+                () =>
+                Authority.CreateAuthorityForRequestAsync(_testRequestContext, s_ppeAuthority, null).Result);
```

Comment:
Now that this is async the test ended up failing. This was the easiest way I found to assert the original conditions when using async in a lambda method. When I tried using the await Authority.CreateAuthorityForRequestAsync() the exception was not thrown and when using .Result the exception is aggregate so I just decided to go with that.

Is there a better way?